sympy/sets/tests/test_sets.py

```diff
@@ -1597,6 +1600,42 @@ def test_DisjointUnion_len():
     assert len(DisjointUnion(S.EmptySet, S.EmptySet, FiniteSet(x, y, z), S.EmptySet)) == 3
     raises(ValueError, lambda: len(DisjointUnion(Interval(0, 1), S.EmptySet)))
 
+def test_SetKind_ProductSet():
+    assert ProductSet(FiniteSet(Matrix([1, 2])), FiniteSet(Matrix([1, 2]))).kind is SetKind(
+    TupleKind(MatrixKind(NumberKind), MatrixKind(NumberKind)))
+    assert ProductSet(
+    Interval(1, 2), FiniteSet(Matrix([1, 2]))).kind is SetKind(TupleKind(NumberKind, MatrixKind(NumberKind)))
+
+def test_SetKind_Interval():
+    assert Interval(1, 2).kind is SetKind(NumberKind)
+
+def test_SetKind_EmptySet_UniversalSet():
+    assert S.UniversalSet.kind is SetKind()
```

Comment:
This one should be `SetKind(UndefinedKind)` I think.

Reply:
I thought `EmptySet` and `UniversalSet` are kind of same, so both should have `SetKind()`